src/org/mozilla/javascript/SecurityBridge_SecurityManager.java

```diff
@@ -0,0 +1,103 @@
+/* -*- Mode: java; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.javascript;
+
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.AllPermission;
+import java.security.Permission;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+
+/**
+ * Code moved from {@link SecurityUtilities}. This implementation makes use of {@link
+ * SecurityManager} and {@link AccessController} and so on, which is deprecated with JDK17 (see <a
+ * href='https://openjdk.java.net/jeps/411'>JEP411</a>) - so all related 'java.security' stuff
+ * should be routed over this class, so that it could be easily replaced by an other implementation
+ * like {@link SecurityBridge_NoOp}.
+ *
+ * <p>This implementation should be work up to JDK17
+ *
+ * @author Attila Szegedi
+ * @author Roland Praml, FOCONIS AG
+ */
+@Deprecated
+public class SecurityBridge_SecurityManager implements SecurityBridge {
+    private static final Permission allPermission = new AllPermission();
+    /**
+     * Retrieves a system property within a privileged block. Use it only when the property is used
+     * from within Rhino code and is not passed out of it.
+     *
+     * @param name the name of the system property
+     * @return the value of the system property
+     */
+    @Override
+    public String getSystemProperty(final String name) {
+        return AccessController.doPrivileged(
+                new PrivilegedAction<String>() {
+                    @Override
+                    public String run() {
+                        return System.getProperty(name);
+                    }
+                });
+    }
```

Comment:
The `getSystemProperty()` method is public and belongs to a public class in an exported package (I mean once Rhino is modularised), allowing the retrieval of system properties with the privileges of Rhino. The comment says "Use it only when the property is used from within Rhino code and is not passed out of it" but the code is not enforcing it: any code with access to the classes in the classpath/modulepath would be allowed to do that.

Probably not a big deal, but I wonder if the security bridge classes could be made package-visible.

Reply:
This could be really a problem